tsp-output/@azure-tools/typespec-csharp/azoai/AzureOpenAI/AzureChatClient.cs

```diff
@@ -0,0 +1,107 @@
+        var value = ModelReaderWriter.Read<CreateChatCompletionResponse>(result.GetRawResponse().Content)!;
+        return ClientResult.FromValue(value, result.GetRawResponse());
+    }
+
+    public override ClientResult<CreateChatCompletionResponse> CreateChatCompletion(CreateChatCompletionRequest createChatCompletionRequest, CancellationToken cancellationToken = default)
+    {
+        Argument.AssertNotNull(createChatCompletionRequest, nameof(createChatCompletionRequest));
+
+        using BinaryContent content = createChatCompletionRequest.ToBinaryContent();
+        ClientResult result = CreateChatCompletion(createChatCompletionRequest.Model.ToString(), content, context: default);
+        var value = ModelReaderWriter.Read<CreateChatCompletionResponse>(result.GetRawResponse().Content)!;
+        return ClientResult.FromValue(value, result.GetRawResponse());
+    }
+
+    public override Task<ClientResult> CreateChatCompletionAsync(BinaryContent content, RequestOptions context = null)
+    {
+        // Note, that we can later remap the values from the 3rd party client format to the 
+        // Azure client format, but this has a perf cost and it's an improvement that can 
+        // come later.
+        throw new InvalidOperationException("Improperly formatted content -- Azure service requires different format. " +
+            $"Please consult the REST API documentation and call the '{nameof(CreateChatCompletion)}' method overload that " +
+            "takes a 'model' parameter.");
+    }
+
+    public override ClientResult CreateChatCompletion(BinaryContent content, RequestOptions context = null)
+    {
+        throw new InvalidOperationException("Improperly formatted content -- Azure service requires different format. " +
+            $"Please consult the REST API documentation and call the '{nameof(CreateChatCompletion)}' method overload that " +
+            "takes a 'model' parameter.");
+    }
+
+    public ClientResult CreateChatCompletion(string model, BinaryContent content, RequestOptions context = null)
+    {
+        Argument.AssertNotNull(model, nameof(model));
+        Argument.AssertNotNull(content, nameof(content));
+
+        using PipelineMessage message = CreateCreateChatCompletionRequest(model, content, context);
+        return ClientResult.FromResponse(Pipeline.ProcessMessage(message, context));
+    }
+
+    public async Task<ClientResult> CreateChatCompletionAsync(string model, BinaryContent content, RequestOptions context = null)
+    {
+        Argument.AssertNotNull(model, nameof(model));
+        Argument.AssertNotNull(content, nameof(content));
+
+        using PipelineMessage message = CreateCreateChatCompletionRequest(model, content, context);
+        return ClientResult.FromResponse(await Pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false));
+    }
+
+    private PipelineMessage CreateCreateChatCompletionRequest(string model, BinaryContent content, RequestOptions context)
```

Comment:
Implements private request-creation helper that creates request with Azure-specific URI and path parameters.  Note here that request content has already been serialized by the model types.